pd/src/app.rs

```diff
@@ -423,13 +425,20 @@ impl App {
                 let current_base_rate = state.base_rate_data(current_epoch.index).await?;
                 let current_rates = state.rate_data(current_epoch.index).await?;
 
+                let mut staking_token_supply = state
+                    .asset_lookup(STAKING_TOKEN_ASSET_ID.encode_to_vec())
+                    .await?
+                    .map(|info| info.total_supply)
+                    .unwrap();
```

Comment:
non-blocking: in a case like this where the code is going to unwrap the option in any case, I think it's slightly cleaner to do so earlier and then work with the inner value directly, rather than mapping on the inner value and then unwrapping.